### DISCUSSION TITLE
Formatting

### DISCUSSION THREAD

**Discussion by Jexly0x** [telegraf/telegraf#1627] [Q&A] [answered]
Hello. I have a simple telegram bot. There is a group to which I write messages, and the bot takes these messages from the channel and copies them into itself. If I write a message to the channel in bold text or any formatting available, the bot repeats itself without formatting. How can I implement it?

![Photo](https://user-images.githubusercontent.com/97974591/169888583-7eeaf2c2-8e9d-484e-b671-a286635c6c66.png)

**Comment by hypernova7** [comment, marked as answer]
Copy the entities, like this

```javascript
bot.on('message', ctx => {
  const { entities, text } = ctx.message
  ctx.reply(text, {
    entities,
    parse_mode: 'Markdown'  // use the same parse_mode that you used in the original message
  })
})
```

**Reply by Jexly0x** [reply]
I use the channel_post action and send a message using bot.telegram.sendMessage(), how do I make the text bold in that case?

![ph](https://user-images.githubusercontent.com/97974591/170001027-600eea39-d5c6-4603-b3b0-94baed3688fd.jpeg)

**Reply by MKRhere** [reply]
```TS
bot.on("channel_post", ctx => {
	if ("text" in ctx.channelPost) {
		// channelPost is a text message
		const { text, entities } = ctx.channelPost;
		bot.telegram.sendMessage(id, text, { entities });
	}
});
```

**Reply by Jexly0x** [reply]
Thank you. I'll test it soon and get back to you. Perhaps you know the answer to my other questions? :D

**Comment by BaalDev-Solutions** [comment]
Does it work? I am trying the same solutions in Telegraf version ^4.9.2 but it does not work. I am unable to get to use the entities when I copy a message

**Reply by BaalDev-Solutions** [reply]
I tried to copy and send the message to my bot with :
`ctx.reply(text, { entities, parse_mode: "Markdown"})
ctx.reply(text, { entities, parse_mode: "HTML"})
ctx.replyWith Markdown(text, { entities })
ctx.replyWith Markdown V2(text, { entities })
`

Also I tried to send a message to a user with:
`await temp_bot.telegram.sendMessage(userId, text, { entities} );
`

Entities are in the next format: 
` entities = { offset: 37, length: 13, type: 'bold' }
`

Does anyone can help me?

**Reply by MKRhere** [reply]
You must not use any `parse_mode` with entities. One of:

```TS
ctx.reply(text, { entities });
ctx.sendMessage(text, { entities });
ctx.telegram.sendMessage(otherChatId, text, { entities });
```

Or maybe you just want

```TS
ctx.copyMessage(otherChatId);
```

**Reply by BaalDev-Solutions** [reply]
@MKRhere, do you know if this is a bug introduced in a later release?

**Reply by MKRhere** [reply]
It is not a bug. If you specify a parse_mode like Markdown or HTML, the entities will be ignored.
